tools/please_pex/pex/pex_main.py

```diff
@@ -187,22 +186,18 @@ def _find_all_distributions(self, module_dir):
 
     def find_spec(self, name, path, target=None):
         """Implements abc.MetaPathFinder."""
-        loader = self.find_module(name, path)
-        if loader is None:
-            return None
-        return spec_from_loader(name, loader)
-
-    def find_module(self, fullname, path=None):
-        """Attempt to locate module. Returns self if found, None if not."""
-        if fullname.startswith(self.prefix):
-            return self
+        if name.startswith(self.prefix):
+            return spec_from_loader(name, self)
 
-    def load_module(self, fullname):
+    def create_module(self, spec):
         """Actually load a module that we said we'd handle in find_module."""
-        module = import_module(fullname[len(self.prefix):])
-        sys.modules[fullname] = module
+        module = import_module(spec.name[len(self.prefix):])
```

Comment:
I think maybe this is mildly not nice (it kind of does everything in `create_module` and leaves nothing for `exec_module`) but I can't find a convenient equivalent to break it into the two parts.